### PR TITLE
[BIOMAGE-1155] [BIOMAGE-1164] Fix volcano plot issues

### PR DESCRIPTION
**Pull request by robioson**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1155
https://biomage.atlassian.net/browse/BIOMAGE-1164
#### Link to staging deployment URL 
https://ui-stealthy-bronze-angelfish.scp-staging.biomage.net/
#### Anything else the reviewers should know about the changes here
The y-axis spec referred to invalid config properties which prevented it from showing up. The reviewer needs to check that y-axis title and values match up (in other words that the values that now display on the y-axis look correct).

While testing the fix Oliver and I noticed some more problems with the volcano plot so I fixed them here too.

When fixing the second part of the ticket ("All" not being selectable) I also fixed a validation issue in the differential expression form.
# Changes
### Code changes
- Got y-axis to show up
- Removed insignificant and unknown gene statuses
- Renamed statuses to be presentable
- Fixed range sliders (set min and max to default values)
- Implemented option for auto range

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/__test__/redux/reducers/componentConfig/__snapshots__/updateConfig.test.js.snap [hunk 1]
```diff
@@ -102,7 +102,7 @@ Object {
         "enabled": true,
         "position": "top-right",
       },
-      "logFoldChangeDomain": 20,
+      "logFoldChangeDomain": 0,
```
Comment (cosa65) [inline src/__test__/redux/reducers/componentConfig/__snapshots__/updateConfig.test.js.snap:105]: why was this changed?

**Comment by cosa65** [comment]
Currently the log2 fold change value can be set as high as we want, to the point that it breaks the plot, as you can see here
Screen Shot 2021-07-09 at 13 01 38

I am thinking that perhaps we could add some kind of maximum value to it based on the values that we receive?
There are some examples of implementations of this in the doublet filter's probability threshold (although in this case I guess the maximum value should be calculated on receiving the result from the worker).

Also, do we want to allow negative values? Because right now we are allowing them, for example I just wrote -100 and got this:
Screen Shot 2021-07-09 at 13 04 15

In doublet filter the component `SliderWithInput` is handling these updates in a way that helps avoid having these kind of problems (a value being set when it is over/below the allowed range). In my opinion I would look into either using this component (in case having a slider is useful here) or refactoring it so that you can use only the input with the max/min values working correctly.

File: src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx [hunk 2]
```diff
@@ -76,8 +75,8 @@ const VolcanoPlot = ({ experimentId }) => {
       controls: [{
         name: 'volcanoDimensions',
         props: {
-          xMax: Math.round(spec.xMax),
-          yMax: Math.round(spec.maxNegativeLogpValue) + 2,
+          xMax: 20,
```
Comment (cosa65) [inline src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx:78]: Why is this value now hardcoded at 20?

File: src/__test__/redux/reducers/componentConfig/__snapshots__/updateConfig.test.js.snap (hunk 1 above)
Comment (robioson) [reply src/__test__/redux/reducers/componentConfig/__snapshots__/updateConfig.test.js.snap:105]: 0 means "auto". Also, 20 was _huge_ as a default and seemed random.

File: src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx (hunk 2 above)
Comment (robioson) [reply src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx:78]: I explained this in the commit description; basically, `xMax` is almost the same as what auto sets, which meant the slider offered basically no adjustment with it as a maximum value. Maybe it would be better to set it here to `xMax * 10`.

**Review by cosa65 — APPROVED** [review]
